eng/CodeChecks.ps1

```diff
@@ -1,82 +1,28 @@
-#requires -version 5
+#Requires -Version 6.0
 
-[CmdletBinding()]
-param (
-    [Parameter(Position=0)]
-    [ValidateNotNullOrEmpty()]
-    [string] $ServiceDirectory
-)
+param([switch]$noReset)
 
 $ErrorActionPreference = 'Stop'
 Set-StrictMode -Version 1
 
-$repoRoot = Resolve-Path "$PSScriptRoot/.."
+. (Join-Path $PSScriptRoot 'DownloadSharedSource.ps1')
+. (Join-Path $PSScriptRoot 'Generate.ps1') -noRun
 
-[string[]] $errors = @()
+try
+{
+    Write-Host 'Downloading shared source files...'
+    Invoke-DownloadSharedSource
 
-function LogError([string]$message) {
-    if ($env:TF_BUILD) {
-        Write-Host "##vso[task.logissue type=error]$message"
-    }
-    Write-Host -f Red "error: $message"
-    $script:errors += $message
-}
-
-function Invoke-Block([scriptblock]$cmd) {
-    $cmd | Out-String | Write-Verbose
-    try
-    {
-        & $cmd
-
-        # Need to check both of these cases for errors as they represent different items
-        # - $?: did the powershell script block throw an error
-        # - $lastexitcode: did a windows command executed by the script block end in error
-        if ((-not $?) -or ($lastexitcode -ne 0)) {
-            if ($error -ne $null)
-            {
-                Write-Warning $error[0]
-            }
-            throw "Command failed to execute: $cmd"
-        }
-    }
-    catch
+    if (-not $noReset)
     {
-        LogError $_
-    }
-}
-
-try {
-    Write-Host "Downloading files"
-    Invoke-Block {
-        & $PSScriptRoot\DownloadSharedSource.ps1 @script:PSBoundParameters
-    }
-
-    Write-Host "Generate test clients"
-    Invoke-Block {
-        & $PSScriptRoot\Generate.ps1 @script:PSBoundParameters
+        Invoke-AutoRest '--reset'
     }
 
-    Write-Host "git diff"
```

Comment:
What verifies that code is up-to-date now that you removed this part?

Reply:
That's my mistake. I had it in my commented code and removed it accidentally.